internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -296,6 +297,33 @@ func (e *BootstrapEnvStackConfig) ToEnv(stack *cloudformation.Stack) (*config.En
 	}, nil
 }
 
+func (e *EnvStackConfig) cdnConfig() template.CDNConfig {
+	var (
+		enabled      bool
+		prefixListId *string
+	)
+
+	if e.in.Mft != nil {
+		// CDN Config specified by AdvancedCDNConfig
+		if !e.in.Mft.CDN.CDNConfig.IsEmpty() {
+			enabled = true
+			prefixListId = e.in.PrefixListID
+		}
+
+		// CDN Config specified by basic boolean enabler
+		if e.in.Mft.CDN.EnableCDN != nil {
+			if *e.in.Mft.CDN.EnableCDN {
+				enabled = true
+			}
+		}
```

Comment:
I know your current implementation is the optimal one. However, since it doesn't really take much longer (could be ignored) and might make it more readable and less repetitive code we could do
```suggestion
                enable = e.in.Mft.CDN.CDNEnabled()
                if !e.in.Mft.CDN.CDNConfig.IsEmpty() {
                    prefixListId = e.in.PrefixListID
                }
```
since stack pkg doesn't really care how mft package is implemented